lte/gateway/c/core/oai/tasks/nas/emm/Attach.c

```diff
@@ -2594,82 +2594,114 @@ static int emm_attach_update(
   OAILOG_FUNC_RETURN(LOG_NAS_EMM, RETURNok);
 }
 
-void proc_new_attach_req(struct ue_mm_context_s* ue_context_p) {
+void proc_new_attach_req(
+    mme_ue_context_t* const mme_ue_context_p,
+    struct ue_mm_context_s* ue_context_p) {
   OAILOG_FUNC_IN(LOG_NAS_EMM);
 
+  hashtable_rc_t hash_rc = HASH_TABLE_OK;
   OAILOG_INFO(
       LOG_NAS_EMM,
       "Process new Attach Request for ue_id " MME_UE_S1AP_ID_FMT "\n",
       ue_context_p->mme_ue_s1ap_id);
-  new_attach_info_t* attach_info = ue_context_p->emm_context.new_attach_info;
-  if (!attach_info) {
-    OAILOG_ERROR_UE(
-        LOG_NAS_EMM, ue_context_p->emm_context._imsi64,
-        "New attach request within EMM context is null \n");
-    OAILOG_FUNC_OUT(LOG_NAS_EMM);
-  }
-
+  new_attach_info_t attach_info = {0};
+  memcpy(
+      &attach_info, ue_context_p->emm_context.new_attach_info,
+      sizeof(new_attach_info_t));
+  free_wrapper((void**) &ue_context_p->emm_context.new_attach_info);
   /* The new Attach Request is received in s1ap initial ue message,
-   * So release previous Attach Request's contexts */
-  if (attach_info->is_mm_ctx_new) {
-    ue_context_p->ue_context_rel_cause = S1AP_NAS_DETACH;
-    /* In case of Ue initiated explicit IMSI Detach or Combined EPS/IMSI detach
-       Do not send UE Context Release Command to eNB before receiving SGs IMSI
-       Detach Ack from MSC/VLR */
-    if (ue_context_p->sgs_context != NULL) {
-      if ((ue_context_p->sgs_detach_type ==
-           SGS_EXPLICIT_UE_INITIATED_IMSI_DETACH_FROM_NONEPS) ||
-          (ue_context_p->sgs_detach_type ==
-           SGS_COMBINED_UE_INITIATED_IMSI_DETACH_FROM_EPS_N_NONEPS)) {
-        OAILOG_FUNC_OUT(LOG_NAS_EMM);
-      } else if (
-          ue_context_p->sgs_context->ts9_timer.id ==
-          MME_APP_TIMER_INACTIVE_ID) {
-        /* Notify S1AP to send UE Context Release Command to eNB or free
-         * s1 context locally.
-         */
+   * So release previous Attach Request's contexts
+   */
+  if (attach_info.is_mm_ctx_new) {
+    if (ue_context_p->ecm_state == ECM_IDLE) {
+      mme_remove_ue_context(mme_ue_context_p, ue_context_p);
```

Comment:
I would suggest adding a debug (maybe info? level) log here to specify UE context is removed due to being on idle mode